sentinel-adapter/sentinel-api-gateway-adapter-common/src/main/java/com/alibaba/csp/sentinel/adapter/gateway/common/slot/GatewayFlowSlot.java

```diff
@@ -27,13 +27,13 @@
 import com.alibaba.csp.sentinel.slots.block.flow.param.ParamFlowException;
 import com.alibaba.csp.sentinel.slots.block.flow.param.ParamFlowRule;
 import com.alibaba.csp.sentinel.slots.block.flow.param.ParameterMetricStorage;
-import com.alibaba.csp.sentinel.spi.SpiOrder;
+import com.alibaba.csp.sentinel.spi.Spi;
 
 /**
  * @author Eric Zhao
  * @since 1.6.1
  */
-@SpiOrder(-4000)
+@Spi(order = -4000)
```

Comment:
I'm a little confused here,  not very sure and use `-4000` temporarily.
There are some constants of slot order definitions In `Constants` class of sentinel-core module:
```java
/**
* Order of default processor slots
*/
public static final int ORDER_NODE_SELECTOR_SLOT = -10000;
public static final int ORDER_CLUSTER_BUILDER_SLOT = -9000;
public static final int ORDER_LOG_SLOT = -8000;
public static final int ORDER_STATISTIC_SLOT = -7000;
public static final int ORDER_AUTHORITY_SLOT = -6000;
public static final int ORDER_SYSTEM_SLOT = -5000;
// order of GatewayFlowSlot -4000
// order of ParamFlowSlot   -3000
public static final int ORDER_FLOW_SLOT = -2000;
public static final int ORDER_DEGRADE_SLOT = -1000;
```
If use constant, which is better?

1. 
```java
public static final int ORDER_GATEWAY_FLOW_SLOT = -4000;
public static final int ORDER_PARAM_FLOW_SLOT = -3000;

@Spi(order = Constants.ORDER_SYSTEM_SLOT)
public class GatewayFlowSlot {
```

2.
```java
No constant for ORDER_GATEWAY_FLOW_SLOT, ORDER_PARAM_FLOW_SLOT
@Spi(order = Constants.ORDER_SYSTEM_SLOT + 1000)
public class GatewayFlowSlot {
```

Reply:
> ```java
> Constants.ORDER_SYSTEM_SLOT
> ```

Yeah whatever you want.

If you want to make it reference-able just replace it with a constant in gateway package;
If you want to express some kinds of relationship with existing slots just use relative calculation to existed constants;
If you insist on pure definition just leave it unchanged.

Reply:
These three ways speak my heart. I've been struggling with this for a long time.
Now I think the third way is not very good. Maybe one or two is better.
For a selection difficulty person, could you please help to make a good choice?

Reply:
> These three ways speak my heart. I've been struggling with this for a long time.
> Now I think the third way is not very good. Maybe one or two is better.
> For a selection difficulty person, could you please help to make a good choice?

Go write a random test and make result confirmed.

Reply:
Got it! Using the first way, since it expresses the order of all default slot of sentinel clearly. Maybe the second way is a good choice for user custom slot.